src/structures/Role.js

```diff
@@ -82,6 +82,44 @@ class Role extends Base {
      * @type {boolean}
      */
     this.deleted = false;
+
+    /**
+     * The tags this role has
+     * @type {?Object}
+     * @property {?Snowflake} botID The id of the bot this role belongs to
+     * @property {?Snowflake} integrationID The id of the integration this role belongs to
+     * @property {?boolean} premiumSubcriberRole Whether this is the guilds premium subscription role
+     */
+    this.tags = data.tags ? {} : null;
+    if (data.tags) {
+      if (Reflect.has(data.tags, 'bot_id')) {
+        this.tags.botID = data.tags.bot_id;
+      }
+      if (Reflect.has(data.tags, 'integration_id')) {
+        this.tags.integrationID = data.tags.integration_id;
+      }
+      if (Reflect.has(data.tags, 'premium_subscriber')) {
+        this.tags.premiumSubscriberRole = true;
+      }
+    }
+  }
+
+  /**
+   * Whether this role is the guilds premium subscription role
+   * @type {boolean}
+   * @readonly
+   */
+  get isPremiumSubscriberRole() {
+    return this.tags ? this.tags.premiumSubscriberRole : false;
+  }
```

Comment:
this would return undefined on a role that has tags, but isn't a premium subscriber role